db/collab.go

```diff
@@ -61,3 +90,38 @@ func (d *DB) CreateCollab(c echo.Context) error {
 
 	return c.String(http.StatusCreated, sessionId)
 }
+
+func (d *DB) JoinCollab(c echo.Context) error {
+	sessionId := c.Param("id")
+	uid := uuid.New().String() // What will we be using as identifiers?
+	session, ok := sessions[sessionId]
+
+	if !ok {
+		return c.String(http.StatusNotFound, "Session does not exist.")
+	}
+
+	websocket.Handler(func(ws *websocket.Conn) {
+		if err := session.AddConn(uid, ws); err != nil {
+			return
+		}
+
+		for {
+			var msg Message
+			err := websocket.JSON.Receive(ws, &msg)
+			if err != nil {
+				errorMsg := Message{
+					Author: uid,
+					Type:   "ERROR",
+					Body:   err.Error(),
+				}
+				session.Broadcast(errorMsg)
```

Comment:
The linter is complaining a bunch here because these errors should be checked. I realize that many of the functions currently return nil regardless, but there are parts of said functions that *could* return an error. We should likely check them to cover corner cases.

Adding error checking would introduce rightward pressure, though. This receipt-process-send routine seems liftable so I recommend introducing some form of error handling for the functions and moving the routine to its own function.

Reply:
Hi, I'm not sure how I can propagate the errors upward from inside the web socket function. It does not seem like ServeHTTP allows for an error to be returned. Am I missing something?